packages/test/test-utils/src/opProcessingController.ts

```diff
@@ -101,26 +137,34 @@ class DeltaManagerMonitor extends DeltaManagerToggle {
             // Clear the pending count and start anew
             this.pendingCount = 0;
             this.firstClientSequenceNumber = -1;
+            this.lastOutbound = undefined;
         });
         deltaManager.outbound.on("op", this.outbound.bind(this));
         deltaManager.inbound.on("push", this.inbound.bind(this));
     }
 
     public get latestSequenceNumber() {
-        return this._latestSequenceNumber;
+        return this.deltaManager.lastSequenceNumber;
     }
 
     public hasPendingWork() {
-        // If we are not connected, we assume that we are trying to, and classify as pending work
-        return this.clientId === undefined || this.pendingCount !== 0;
+        return !this.deltaManager.disposed
+            && (this.pendingWriteConnection || this.pendingCount !== 0);
```

Comment:
RE pendingWriteConnection: There is really nothing requiring IRuntime to resubmit an op on a nack.
I.e. runtime may chose to not submit anything on reconnection. Great example of that would be potential improvement in Consensus Registry DDS, where it would not resent an op where it knows for sure it would be ignored by all clients (today we resend everything, i.e. we do not do any optimizations).

Can we make this logic more generic and not rely on any particular behavior of IRuntime around nacks?

Reply:
i poked around a bite here when making this change, but this transition is hard to track reliably. open to other options if you have ideas about how to make this better